server/gfmd/auth_uid.c

```diff
@@ -360,7 +360,7 @@ auth_uid_to_global_username_sasl(void *closure,
 		return (GFARM_ERR_AUTHENTICATION);
 
 	giant_lock();
-	u = user_tenant_lookup(auth_user_id);
+	u = user_lookup_auth_id(GFARM_AUTH_USER_ID_TYPE_SASL, auth_user_id);
```

Comment:
how about exporting enum auth_user_id_type in user.h, and passing AUTH_USER_ID_TYPE_SASL instead of GFARM_AUTH_USER_ID_TYPE_SASL as the first parameter of user_lookup_auth_id(), to avoid the dynamic string->enum conversion by gfarm_auth_user_id_type_from_name() in user_lookup_auth_id()?